pkg/controller/controller.go

```diff
@@ -84,6 +84,8 @@ type ErrorDetails struct {
 	Message     string
 }
 
+var iapHeaderName = "x-goog-authenticated-user-email"
```

Comment:
Should we be validating these headers? https://cloud.google.com/iap/docs/signed-headers-howto

Reply:
Yeah I was thinking about this, I decided not to for the following reasons:

1. Given your recent decoupling of the email for subject and letting the downstream service validate reduces the emphasis on the validity of the email 
2. AFIK local testing wouldnt be possible given we would need a valid JWT from IAP? [They have this](https://cloud.google.com/iap/docs/query-parameters-and-headers-howto#testing_jwt_verification) for testing but its through query params instead of headers.

Reply:
I'm fine not doing it in this PR (I thought we were going to add validation earlier) but we should prioritize adding it.

Reply:
I agree this complicates local testing. I also agree with Chen is should be in a different PR. Let's make an issue to design out validation for gRPC and IAP headers, documenting the need for a good local dev story (e.g. maybe we skip validation when `DEV_MODE=true`).

Reply:
Ack ill create an issue.

Reply:
gRPC is tricky if it's on cloud run. Cloud run will redact the JWT signature so we can no longer validate it.

Reply:
https://github.com/abcxyz/jvs/issues/203

Reply:
@yolocs I think they fixed that actually!

Reply:
Interesting! Do you have a reference? Haven't tried it for real but the doc still says so https://cloud.google.com/run/docs/troubleshooting#signature-removed

Reply:
We might be talking about two different things. I'll ping you on chat.